packages/expo-network/ios/EXNetwork/EXNetwork.m

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXNetwork/EXNetwork.h>
+#import <SystemConfiguration/SystemConfiguration.h>
+
+#import <ifaddrs.h>
+#import <arpa/inet.h>
+
+@interface EXNetwork ()
+
+@property (nonatomic, weak) UMModuleRegistry *moduleRegistry;
+@property (nonatomic) SCNetworkReachabilityRef reachabilityRef;
+@property (nonatomic) SCNetworkReachabilityFlags lastFlags;
+@property (nonatomic) NSString *type;
+
+@end
+
+@implementation EXNetwork
+
+// Creates a new "blank" state
+- (instancetype)init
+{
+  self = [super init];
+  if (self) {
+    _type = EXNetworkTypeUnknown;
+  }
+  return self;
+}
+
+UM_EXPORT_MODULE(ExpoNetwork);
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _moduleRegistry = moduleRegistry;
+}
+
+
+UM_EXPORT_METHOD_AS(getMacAddressAsync,
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  //some iOS privacy issues
+  NSString *address = @"02:00:00:00:00:00";
+  resolve(address);
+}
+
+UM_EXPORT_METHOD_AS(getIpAddressAsync,
+                    getIpAddressAsyncWithResolver:(UMPromiseResolveBlock)resolve rejecter:(UMPromiseRejectBlock)reject)
+{
+  NSString *address = @"0.0.0.0";
+  struct ifaddrs *interfaces = NULL;
+  struct ifaddrs *temp_addr = NULL;
+  int error = 0;
+  // retrieve the current interfaces - On success, returns 0; on error, -1 is returned, and errno is set appropriately.
```

Comment:
what is `errno`? if there's more information about an error that could happen we probably want to capture it and pass it back to the user in the promise rejection.